subprojects/maven/src/test/groovy/org/gradle/api/publish/maven/internal/tasks/MavenPomFileGeneratorTest.groovy

```diff
@@ -336,31 +277,51 @@ class MavenPomFileGeneratorTest extends Specification {
 
     def "applies withXml actions"() {
         when:
-        generator.withXml(new Action<XmlProvider>() {
-            void execute(XmlProvider t) {
-                t.asNode().groupId[0].value = "new-group"
-            }
-        })
-        generator.withXml(new Action<XmlProvider>() {
-            void execute(XmlProvider t) {
-                t.asNode().appendNode("description", "custom-description-ぴ₦ガき∆ç√∫")
-            }
-        })
+        pom.withXml {
+            asNode().groupId[0].value = "new-group"
+        }
+        pom.withXml {
+            asNode().appendNode("description", "custom-description-ぴ₦ガき∆ç√∫")
+        }
 
         then:
-        with (pom) {
+        with (xml) {
             groupId == "new-group"
             description == "custom-description-ぴ₦ガき∆ç√∫"
         }
     }
 
-    private def getPom() {
+    private MavenPomDependencies newDependencies(MavenDependencyInternal dependency) {
+        return new DefaultMavenPomDependencies(
+            ImmutableList.of(dependency), ImmutableList.of(), ImmutableList.of(), ImmutableList.of(),
+            ImmutableList.of(), ImmutableList.of(), ImmutableList.of()
+        )
+    }
+
+    private MavenPomInternal newPom() {
+        MavenPomInternal pom = TestUtil.objectFactory().newInstance(
+            DefaultMavenPom.class,
+            TestUtil.objectFactory(),
+            strategy
+        )
+
+        pom.projectIdentity.groupId.set("group-id")
+        pom.projectIdentity.artifactId.set("artifact-id")
+        pom.projectIdentity.version.set("1.0")
+
+        pom.dependencies.set(DefaultMavenPomDependencies.EMPTY)
+        pom.getWriteGradleMetadataMarker().set(true)
+
+        return pom
+    }
+
+    private def getXml() {
         return new XmlSlurper().parse(pomFile);
     }
 
     private TestFile getPomFile() {
         def pomFile = testDirectoryProvider.testDirectory.file("pom.xml")
-        generator.writeTo(pomFile)
+        generator.generateSpec(pom).writeTo(pomFile)
```

Comment:
💅 Could we name this method `writePomFile` instead?  It's more than a getter.

Reply:
Renamed